src/components/page-configuration/build-secrets-toggler.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+import AsyncResource from '../async-resource/simple-async-resource';
+import Panel from '../panel';
+import Toggler from '../toggler';
+import SecretStatus from '../secret-status';
+
+import useGetBuildSecrets from '../page-build-secret/use-get-build-secrets';
+
+import * as routing from '../../utils/routing';
+
+const BuildSecretsToggler = props => {
+  const getBuildSecretsState = useGetBuildSecrets(props.appName);
+  const data = getBuildSecretsState.data;
+
+  return (
+    <Panel>
```

Comment:
not sure if its worth it, but maybe you can generalise `build-secret-toggler.js` and `image-hubs-toggler.js` and use a common component